tests/test_username_api.py

```diff
@@ -0,0 +1,249 @@
+import unittest
+import json
+import yaml
+import string
+import random
+import pytest
+import username_api
+
+data = yaml.load(open('tests/test_data.yml'))
+websites = yaml.load(open('websites.yml'))
+
+invalid_username = '$very%long{invalid}user(name)'
+
+def generate_random_username(length):
+  return ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase \
+    + string.digits) for _ in range(length))
+
+def assert_response(app, website, user, status):
+  expected = get_expected_response(website, user, status)
+  actual = get_response(app, website, user)
+  assert expected == actual
+
+def get_response(app, website, user):
+  resp = app.get('/check/{}/{}'.format(website, user))
+  return json.loads(resp.get_data())
+
+def get_expected_response(website, user, status):
+  return {
+    'possible': True,
+    'status': status,
+    'url': websites['urls'].get(website, 'https://{w}.com/{u}').format(
+      w=website,
+      u=user
+    )
+  }
+
+class TestUsernameApi(object):
+  @classmethod
+  def setup_class(self):
+    self.app = username_api.app.test_client()
+    self.data = data
+
+  @pytest.mark.parametrize('user', data['github']['taken_usernames'])
+  def test_taken_github_username(self, user):
```

Comment:
the website can also be parameterised, but maybe we leave that for another task.